### PR TITLE
feat: upgrade to Flutter 3.24 and Yaru 5.x

### PR DESCRIPTION
**Pull request by d-loose**
I've pointed out the non-trivial code changes below, everything else is due to https://github.com/ubuntu/yaru.dart/pull/836 (`yaru_{widgets,icons}.dart` merged into `yaru.dart`) or due to the `require_trailing_commas` linter rule added to `ubuntu_lints`.

Please do have a closer look at the changes in `melos.yaml`, `.github/` and `snap/`, though :)

UDENG-4599

### CODE REVIEW

File: packages/firmware_updater/lib/src/widgets/device_panel_list.dart
Comment (d-loose) [inline packages/firmware_updater/lib/src/widgets/device_panel_list.dart:1]: Yaru deprecated the `YaruExpansionPanelList` widget used here - but it seems like `DevicePanelList` isn't being used anywhere anymore, so I removed it entirely.

File: packages/firmware_updater/lib/detail_page.dart [hunk 1]
```diff
@@ -44,25 +45,24 @@ class _DetailPageState extends State<DetailPage> {
...
-                .select<DeviceStore, bool>((store) => store.showReleases))
-              const MaterialPage(
-                child: ReleasePage(),
-              )
-          ],
-          onPopPage: (route, result) {
-            return route.didPop(result);
-          },
+          pages: pages,
+          onDidRemovePage: pages.remove,
```
Comment (d-loose) [inline packages/firmware_updater/lib/detail_page.dart:65]: See https://docs.flutter.dev/release/breaking-changes/navigator-and-page-api

File: melos.yaml [hunk 2]
```diff
@@ -1,64 +1,73 @@
 name: firmware_updater_workspace
+sdkPath: .fvm/flutter_sdk
 
 packages:
-  - packages/**
+  - packages/*
```
Comment (sminez) [inline melos.yaml:5]: what's the story with changing the glob here? Was this not working previously or is it a change on the melos side that we're following?

**Review by sminez — APPROVED** [review]
Looking good, thank you for this!

Approving assuming that the change I've asked about in the `melos.yaml` isn't something we need to double check. So long as that's fine this looks good to merge :+1:

**Pull request merged**